### PR TITLE
06_timers: task solution

### PR DESCRIPTION
**Pull request by ASynenko**
Kmodule with sysfs attributes for time of/since last call read.
Uspace utility prints current UTC time

### CODE REVIEW

File: 06_timers/uspace/main.c [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+int main(int argc, char *argv[])
+{
+	struct timespec timespec_var;
+	char buffer[200] = {'\0'};
+
+	timespec_get(&timespec_var, TIME_UTC);
+	strftime(buffer, sizeof(buffer), "%T",
+		gmtime(&timespec_var.tv_sec));
+	printf("Time now in UTC: %s.%09ld UTC\n", buffer,
```
Comment (ReAlUA) [inline 06_timers/uspace/main.c]: One of these `UTC` is redundant

File: 06_timers/kmodule/timer_module.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	time_of_last = ktime_get_ns();
+
+	return sprintf(buf, "t[-1]: %llu.%03llu sec\n", then_sec, then_ns);
+}
+
+static ssize_t rel_show(struct kobject *kobj, struct kobj_attribute *attr,
+					   char *buf)
+{
+	static u64 time_of_last;
```
Comment (ReAlUA) [inline 06_timers/kmodule/timer_module.c]: These two static variables of name `time_of_last` (here and above) do not confuse compiler/linker but may confuse code supporters.

File: 06_timers/kmodule/timer_module.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	u64 then_sec, then_ns;
+
+	then_sec = div64_u64_rem(time_of_last, NSEC_PER_SEC, &then_ns);
+
+	time_of_last = ktime_get_ns();
+
+	return sprintf(buf, "t[-1]: %llu.%03llu sec\n", then_sec, then_ns);
```
Comment (ReAlUA) [inline 06_timers/kmodule/timer_module.c]: For **nano**seconds you must use `%09`

File: 06_timers/kmodule/timer_module.log [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/ # 
+/ # insmod /home/timer_module.ko
```
Comment (ReAlUA) [inline 06_timers/kmodule/timer_module.log:2]: Add a small test script with `sleep` for different times, say, 0.1, 0.33, etc.

**Review by ReAlUA — CHANGES_REQUESTED** [review]

File: 06_timers/uspace/main.c (hunk 1 above)
Comment (ASynenko) [reply 06_timers/uspace/main.c]: Fixed

File: 06_timers/kmodule/timer_module.c (hunk 2 above)
Comment (ASynenko) [reply 06_timers/kmodule/timer_module.c]: Strange. But fixed

File: 06_timers/kmodule/timer_module.c (hunk 3 above)
Comment (ASynenko) [reply 06_timers/kmodule/timer_module.c]: Fixed

File: 06_timers/kmodule/timer_module.c (hunk 2 above)
Comment (ReAlUA) [reply 06_timers/kmodule/timer_module.c]: It looks easy in a simple module with short functions, but in real life it is a place where you can make mistakes when modifying the code.

**Review by ReAlUA — APPROVED** [review]

**Pull request merged**